Firestore/Example/Tests/Integration/API/FIRDatabaseTests.mm

```diff
@@ -1240,4 +1241,79 @@ - (void)testClientCallsAfterShutdownFail {
       NSException, @"The client has already been shutdown.");
 }
 
+- (void)testMaintainsPersistenceAfterRestarting {
+  FIRDocumentReference *doc = [self documentRef];
+  FIRFirestore *firestore = doc.firestore;
+  FIRApp *app = firestore.app;
+  NSString *appName = app.name;
+  FIROptions *options = app.options;
+
+  NSDictionary<NSString *, id> *initialData = @{@"foo" : @"42"};
+  [self writeDocumentRef:doc data:initialData];
+
+  // -clearPersistence() requires Firestore to be shut down. Shutdown FIRApp to emulate the way
+  // an end user would do this.
+  [self shutdownFirestore:firestore];
+  [self.firestores removeObject:firestore];
```

Comment:
Optional: now that this is no longer intrinsic to `-shutdownFirestore:`, is it worth commenting on why this is required?